src/NLog/Conditions/ConditionParser.cs

```diff
@@ -136,7 +136,7 @@ private ConditionMethodExpression ParsePredicate(string functionName)
             }
             catch (Exception exception)
             {
-                if (exception.MustBeRethrown())
+                if (exception.MustBeRethrown("While resolving function '{0}' follwing exception occured: {1}", functionName, exception))
```

Comment:
proposal: **always** add the exception to the end of the message. So no need to to pass exception twice (implicit and explicit). Need a lot of changes.